### PR TITLE
[CC-42]: Add filter mode into service area

### PR DESCRIPTION
**Pull request by jiayuu-u**
##### Link to Jira ticekt here:
[https://leonhard.atlassian.net/browse/CC-42](https://leonhard.atlassian.net/browse/CC-42)

### Descriptions
Add filter mode when franchisee selects the area

### Provide screenshots or videos that the function is working.
image

### Provide screenshots of no warning, error in console.
image

### Developer Checklist

- [x] No merge conflict with the main branch, make sure your code is up to date with main.
- [x] Check PR title is aligned with your ticket, (i.e [CC-xx: Feature] Create pull request template).
- [x] Check your app will run without crashes.
- [x] Check if there are hard coded values, note them if there are anything.
- [x] No unnecessary spaces between lines of code.
- [x] Check you are using camelCase && PascalCase correctly.
- [x] Check you named your code properly.
- [x] Remove unused comments.

### CODE REVIEW

**Comment by NorrisWu0** [comment]
[https://leonhard.atlassian.net/browse/CC-42?atlOrigin=eyJpIjoiMGUzNmNlNzI4OGQ4NGUyYjk0MzRhMmQ0ZGNmMjAzM2EiLCJwIjoiaiJ9](https://github.com/courtcanva/cc-franchise/pull/url)

I can find the ticket via the issue number but, this link doesn't seems to work?

**Comment by jiayuu-u** [comment]
> 

The link is working now

File: src/test/java/com/courtcanva/ccfranchise/utils/SuburbTestHelper.java [hunk 1]
```diff
@@ -13,27 +14,19 @@
...
-        return SuburbGetDto.builder()
-                .sscCode(11344L)
-                .suburbName("East Albury")
-                .postcode(2640)
-                .state(AUState.NSW)
-                .build();
+    public static SuburbListAndFilterModePostDto createSuburbListPostDtoWithIncludeMode() {
+        List<SuburbPostDto> suburbs = new ArrayList<>();
+        suburbs.add(new SuburbPostDto(11344L));
+        suburbs.add(new SuburbPostDto(12287L));
```
Comment (sgz886) [inline src/test/java/com/courtcanva/ccfranchise/utils/SuburbTestHelper.java:21]: can be simplified to
        `List<SuburbPostDto> suburbs = List.of(new SuburbPostDto(11344L), suburbs.add(new SuburbPostDto(12287L))`

File: src/test/java/com/courtcanva/ccfranchise/utils/SuburbTestHelper.java [hunk 2]
```diff
@@ -13,27 +14,19 @@
...
+        return SuburbListAndFilterModePostDto.builder().filterMode(DutyAreaFilterMode.INCLUDE).suburbs(suburbs).build();
     }
 
-
-    public static SuburbListPostDto createSuburbListPostDto() {
+    public static SuburbListAndFilterModePostDto createSuburbListPostDtoWithExcludeMode() {
         List<SuburbPostDto> suburbs = new ArrayList<>();
         suburbs.add(new SuburbPostDto(11344L));
         suburbs.add(new SuburbPostDto(12287L));
```
Comment (sgz886) [inline src/test/java/com/courtcanva/ccfranchise/utils/SuburbTestHelper.java:28]: can be simplified to
`List<SuburbPostDto> suburbs = List.of(new SuburbPostDto(11344L), suburbs.add(new SuburbPostDto(12287L))`

**Review by NorrisWu0 — APPROVED** [review]

**Review by leonandhard — APPROVED** [review]

**Pull request merged**